File: test/container/array/gen_python/gen_array.py
import os
import sys
import json

output_file = "/data/generated_array.json"

# Read and validate COUNT environment variable
count_env = os.getenv("COUNT")
try:
    count = int(count_env)
    if count < 1:
        raise ValueError
except (TypeError, ValueError):
    count = 10  # Default value

# Generate JSON array from 1 to COUNT
numbers = list(range(1, count + 1))

# Write to output file
with open(output_file, "w") as f:
    json.dump(numbers, f)

print(f"JSON array 1...{count} written to {output_file}.")

File: test/container/array/gen_r/gen_array.R
# Load the required library for JSON handling
library(jsonlite)

# Read the COUNT environment variable, convert it to an integer and check its validity
count_env <- Sys.getenv("COUNT")
count <- as.integer(count_env)
if (is.na(count) || count < 0) {
  stop("Invalid COUNT value. Please set COUNT to a non-negative integer.")
}

numbers <- 0:count

write_json(numbers, "/data/generated_array.json", pretty = TRUE)
cat("Numbers written to /data/generated_array.json\n")
